tests/integration/any_charm.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This code snippet is used to be loaded into any-charm which is used for integration tests."""
+import os
+import pathlib
+import signal
+import subprocess
+
+from any_charm_base import AnyCharmBase
+from ingress import IngressRequires
+
+
+class AnyCharm(AnyCharmBase):
+    """Execute a simple web-server charm to test the ingress-proxy relation"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ingress = IngressRequires(
+            self,
+            {"service-hostname": self.app.name, "service-name": self.app.name, "service-port": 80},
+        )
+
+    def update_ingress(self, ingress_config):
+        self.ingress.update_config(ingress_config)
+
+    @staticmethod
+    def start_server(port: int = 80):
+        """Start an HTTP server daemon."""
+        www_dir = pathlib.Path("/tmp/www")
+        www_dir.mkdir(exist_ok=True)
+        ok_file = www_dir / "ok"
+        ok_file.write_text("ok")
+        # We create a pid file to avoid concurrent executions of the http server
+        pid_file = pathlib.Path("/tmp/any.pid")
+        if pid_file.exists():
+            os.kill(int(pid_file.read_text()), signal.SIGKILL)
+            pid_file.unlink()
+        p = subprocess.Popen(
```

Comment:
There is an open PR in the IS Charms Contributing Guide about subprocess usage and error handling. Do you think will be needed here?

https://github.com/canonical/is-charms-contributing-guide/blob/53805351bb7b222e1acb23c9eecc09557f246f3a/CONTRIBUTING.md#subprocess-calls-within-python

Reply:
Since this is for a test, I'm inclined to have something as simple as possible. I think that the documentation about subprocess usage makes more sense for production code

Reply:
The subprocess here is actually to run a daemon/detached HTTP server in the background. It's a little different than the typical usage of subprocess.